core/state/statedb.go

```diff
@@ -989,7 +989,7 @@ func (s *StateDB) CorrectAccountsRoot(blockRoot common.Hash) {
 	if accounts, err := snapshot.Accounts(); err == nil && accounts != nil {
 		for _, obj := range s.stateObjects {
 			if !obj.deleted && !obj.rootCorrected && obj.data.Root == dummyRoot {
-				if account, exist := accounts[crypto.Keccak256Hash(obj.address[:])]; exist && len(account.Root) != 0 {
+				if account, exist := accounts[crypto.Keccak256Hash(obj.address[:])]; exist {
```

Comment:
This is the only place that changes the behavior, others changes in this PR is just code refine.
But how could `len(account.Root) == 0` ?